app/lib/services/import.dart

```diff
@@ -517,6 +583,47 @@ class ImportService {
   }
 
   NoteData? _submit(
+    NoteData document, {
+    required List<List<PadElement>> batchesOfElements,
+    List<DocumentPage> pages = const [],
+    List<Area> areas = const [],
+    bool choosePosition = false,
+    required List<PadElement> elements,
+  }) {
+    final state = _getState();
+    DocumentPage page =
+        state?.page ?? document.getPage() ?? DocumentDefaults.createPage();
+    //cicle
+
+    if (choosePosition && state != null) {
+      for (var batch in batchesOfElements) {
+        if (batch.isNotEmpty || areas.isNotEmpty) {
+          state.currentIndexCubit.changeTemporaryHandler(
```

Comment:
What are you doing there? You override the last element when setting the ImportTool